docs/DeveloperGuide.md

```diff
@@ -2,36 +2,530 @@
 
 ## Acknowledgements
 
```

Comment:
Overall, everything looks great. Really nice design and diagrams. Couldn't be better